### PR TITLE
Configuration commands

### PR DESCRIPTION
**Pull request by JohnMcPMS**
## Change
Adds the configuration commands `configure`, `configure show`, `configure test`, and `configure validate` (although it doesn't actually implement `test` or `validate`).

`configure show` outputs the configuration file that it is given, along with information on the configuration units contained within.  With the current PS DSC processor implementation, an example configuration file (that doesn't actually do anything as it is missing settings):
```YAML
properties:
    configurationVersion: 0.1
    resources:
      - resource: XmlFileContentResource
        directives:
          module: XmlContentDsc
          description: Set XML file contents
      - resource: SecureBoot
        directives:
          module: DellBIOSProvider
          description: Set secure boot options
```
is shown as:
```
> wingetdev configure show ShowDetails.yml
Apply :: XmlFileContentResource
  Set XML file contents
  Module: XmlContentDsc by Raimund Andree [PSGallery]
    Module with DSC Resources for managing XML file content
Apply :: SecureBoot
  Set secure boot options
  Module: DellBIOSProvider by Dell BizClient Team [PSGallery]
    The 'Dell Command | PowerShell Provider' provides native configuration capability of Dell Optiplex, Latitude, Precision, XPS Notebook and Venue 11 systems within PowerShell.
```

`configure` will first show the configuration file, then prompt the user to both review it and accept the responsibility for its actions. `--accept-configuration-agreements` can pre-accept the warning and apply the configuration without prompting. As the configuration is being applied, the individual results will be reported.

All of the configuration actions are run in a separate process to isolate winget.exe from a potential crash. That process is run at the same integrity level as winget.exe though, allowing elevated (admin) configuration actions.

Additionally, there are several minor fixes to things that I saw while developing this.

## Validation
Currently this has only been manually validated (the command workflow; the configuration code has automated tests).  Tests will be backfilled for this, but for now we are trying to get this into selfhost.  As it is an experimental feature and also completely new, this seems like an acceptable, temporary state to be in.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/3036)

### CODE REVIEW

**Comment by denelon** [comment]
Related to:
* https://github.com/microsoft/winget-cli/issues/2845

File: src/Microsoft.Management.Configuration.UnitTests/Helpers/TestConfigurationProcessorFactory.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace Microsoft.Management.Configuration.UnitTests.Helpers
     /// <summary>
     /// A test implementation of IConfigurationProcessorFactory.
```
Comment (msftrubengu) [inline src/Microsoft.Management.Configuration.UnitTests/Helpers/TestConfigurationProcessorFactory.cs:13]: >IConfigurationProcessorFactory [](http://example.com/codeflow?start=33&length=30)

IConfigurationSetProcessorFactory

File: src/Microsoft.Management.Configuration.Processor/Public/ConfigurationSetProcessorFactory.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ namespace Microsoft.Management.Configuration.Processor
     /// ConfigurationProcessorFactory implementation. Does not support out-of-proc.
     /// TODO: change to IConfigurationSetProcessorFactory.
```
Comment (msftrubengu) [inline src/Microsoft.Management.Configuration.Processor/Public/ConfigurationSetProcessorFactory.cs:15]: delete

File: src/ConfigurationRemotingServer/Program.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+using Microsoft.Management.Configuration.Processor;
+using System.Runtime.InteropServices;
+using WinRT;
```
Comment (msftrubengu) [inline src/ConfigurationRemotingServer/Program.cs:3]: I know you'll hate me, but can you add stylecops :D

Comment (JohnMcPMS) [reply src/ConfigurationRemotingServer/Program.cs:3]: 😡

**Comment by JohnMcPMS** [comment]
As is blatantly obvious from the commit count and messages, the build server has been difficult.  While I don't think that the project setup is workable for running tests on the build server yet, I think that we need to be able to carry this through to the internal build to work out the issues for selfhosting purposes.  Those fixes will then be applied here and hopefully enable E2E testing to occur soon.

Since the local builds work, and this is a completely siloed experimental feature, these current shortcomings are acceptable to me.

**Comment by denelon** [comment]
I agree since we're working on preview for an experimental feature.

**Comment by Vysakh131101** [comment]
91738694b55ffddd22b5f895f374e15a71b34492

**Pull request merged**